netbox/dcim/models/mixins.py

```diff
@@ -0,0 +1,47 @@
+from django.core.exceptions import ValidationError
+from django.db import models
+from dcim.choices import *
+from utilities.utils import to_kilograms
+
+
+class DeviceWeightMixin(models.Model):
+    weight = models.DecimalField(
+        max_digits=8,
+        decimal_places=2,
+        blank=True,
+        null=True
+    )
+    weight_unit = models.CharField(
+        max_length=50,
+        choices=DeviceWeightUnitChoices,
+        blank=True,
+    )
+    # Stores the normalized weight (in kilograms) for database ordering
+    _abs_weight = models.DecimalField(
+        max_digits=10,
+        decimal_places=2,
```

Comment:
Should we just store grams instead? A ton (2000lb) is "only" 907,184 grams, so it seems like a PositiveBigIntegerField would be plenty sufficient. (It seems unlikely that we would need milligram precision.)